lib/betamocks/optional_locator.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+require 'rack'
+
+module Betamocks
+  class OptionalLocator
+    attr_writer :env, :endpoint_config
+
+    def initialize(env)
+      @env = env
+      @endpoint_config = Betamocks.configuration.find_endpoint(@env)
+    end
+
+    def generate
+      matched_optional_locator = generate_optional_locator_dir(@endpoint_config[:cache_multiple_responses])
+      return '' unless matched_optional_locator
+
+      matched_optional_locator.gsub(/[^0-9a-z]/i, '')
+    end
+
+    private
+
+    def generate_optional_locator_dir(locator_config)
```

Comment:
Since this is private, do we really need to check if locator_config is passed in? I think it would throw an exception if it was passed in as nil.

Reply:
There are a set of services that don't define the `cache_multiple_responses` field, so even though `@endpoint_config` might not be `nil`, the `@endpoint_config[:cache_multiple_responses]` parameter could be nil.